spinnaker_camera_driver/src/nodelet.cpp

```diff
@@ -621,6 +636,81 @@ class SpinnakerCameraNodelet : public nodelet::Nodelet {
 
             wfov_image->info = *ci_;
 
+
+            #include <cv_bridge/cv_bridge.h>
+
+            #include "spinnaker_camera_driver/SpinnakerCamera.h" // The actual standalone library for the Spinnakers
+            #include "spinnaker_camera_driver/diagnostics.h"
+
+            #include <camera_info_manager/camera_info_manager.h> // ROS library that publishes CameraInfo topics
+            #include <image_transport/image_transport.h> // ROS library that allows sending compressed images
+            #include <opencv2/imgproc/imgproc.hpp>
+            #include <sensor_msgs/CameraInfo.h> // ROS message header for CameraInfo
+            #include <sensor_msgs/image_encodings.h>
+
+            if (histogram_equalization_enabled) {
+              bool success = true;
+
+              cv_bridge::CvImagePtr cv_ptr;
+              try {
+                cv_ptr = cv_bridge::toCvCopy(wfov_image->image,
+                                             wfov_image->image.encoding);
+              } catch (cv_bridge::Exception &e) {
+                ROS_ERROR("cv_bridge exception: %s", e.what());
+                success = false;
+              }
+              if (success) {
+                cv::Mat &image_to_process = cv_ptr->image;
+                if (wfov_image->image.encoding ==
+                    sensor_msgs::image_encodings::RGB8) {
+                  assert(image_to_process.channels() == 3);
+                  cv::Mat ycrcb;
+                  cv::cvtColor(image_to_process, ycrcb, CV_RGB2YCrCb);
+                  std::vector<cv::Mat> channels;
+                  cv::split(ycrcb, channels);
+
+                  static cv::Ptr<cv::CLAHE> clahe = cv::createCLAHE(
+                      histogram_equalization_clahe_clip_limit,
+                      cv::Size(histogram_equalization_clahe_grid_size,
+                               histogram_equalization_clahe_grid_size));
+                  clahe->apply(channels[0], channels[0]);
```

Comment:
One could think of one more option/flag to include standard histogram equalization
```
cv::equalizeHist( src, dst );
```
as I had better results/less artifacts with it in really dark environments.